//
// Copyright 2022 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package psi.sealpir;

message PlaintextsProto {
  repeated bytes plains = 1;
}

message CiphertextsProto {
  repeated bytes ciphers = 1;
}

message SealPirResponseProto {
  repeated CiphertextsProto response_cipher = 1;
}

message SealPirQueryProto {
  repeated CiphertextsProto query_cipher = 1;
}

message SealPirServerProto {
  uint64 raw_db_rows = 1;
  uint64 raw_db_bytes = 2;

  uint32 dim = 4;
  repeated uint64 dim_vec = 5;

  uint32 poly_modulus_degree = 6;
  uint32 logt = 7;

  uint64 partition_num = 8;
  uint64 pt_num = 9;

  repeated PlaintextsProto dbs = 10;
  map<uint64, bytes> galois_keys = 11;
}